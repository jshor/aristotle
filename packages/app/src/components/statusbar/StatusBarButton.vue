<template>
  <button
    class="status-bar-button"
    :disabled="disabled"
    :title="title"
    :style="{ width }"
  >
    <slot />
  </button>
</template>

<script lang="ts">
import { defineComponent } from 'vue'

export default defineComponent({
  name: 'StatusBarButton',
  props: {
    /** Whether or not to disable the button. */
    disabled: {
      type: Boolean,
      default: false
    },
    /** Tooltip-friendly button text. */
    title: {
      type: String,
      default: ''
    },
    /** Style width. */
    width: {
      type: String,
      default: '2em'
    }
  }
})
</script>

<style lang="scss">
.status-bar-button {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100%;
  font-family: var(--font-family);
  color: var(--color-primary);
  background-color: transparent;
  padding: 0 0.5em;
  box-sizing: border-box;
  border: 0;
  cursor: pointer;

  &:disabled {
    opacity: 0.5;
    cursor: default;
  }

  &:not(:disabled) {
    &:hover {
      background-color: var(--color-bg-quaternary);
    }

    &:active {
      background-color: var(--color-bg-tertiary);
    }
  }
}
</style>
