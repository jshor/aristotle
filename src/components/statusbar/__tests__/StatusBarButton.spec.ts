import { shallowMount, VueWrapper } from '@vue/test-utils'
import { ComponentPublicInstance } from 'vue'
import StatusBarButton from '../StatusBarButton.vue'

describe('Status Bar Button', () => {
  let wrapper: VueWrapper

  beforeEach(() => {
    wrapper = shallowMount(StatusBarButton)
  })

  it('should match the snapshot', async () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
