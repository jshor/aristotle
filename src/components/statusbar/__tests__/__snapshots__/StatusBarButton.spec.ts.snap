// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Status Bar Button > should match the snapshot 1`] = `"<button class="status-bar-button" title="" style="width: 2em;"></button>"`;

exports[`Status Bar Button should match the snapshot 1`] = `"<button class="status-bar-button" title="" style="width: 2em;"></button>"`;
